pep-0xxx.rst

```diff
@@ -0,0 +1,727 @@
+PEP: XXX
+Title: TLS Abstract Base Classes
+Version: $Revision$
+Last-Modified: $Date$
+Author: Cory Benfield <cory@lukasa.co.uk>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 17-Oct-2016
+Python-Version: 3.7
+Post-History: 30-Aug-2002
+
+
+Abstract
+========
+
+This PEP would define a standard TLS interface in the form of a collection of
+abstract base classes. This interface would allow Python implementations and
+third-party libraries to provide bindings to TLS libraries other than OpenSSL
+that can be used by tools that expect the interface provided by the Python
+standard library, with the goal of reducing the dependence of the Python
+ecosystem on OpenSSL.
+
+
+Rationale
+=========
+
+In the 21st century it has become increasingly clear that robust and
+user-friendly TLS support is an extremely important part of the ecosystem of
+any popular programming language. For most of its lifetime, this role in the
+Python ecosystem has primarily been served by the `ssl module`_, which provides
+a Python API to the `OpenSSL library`_.
+
+Because the ``ssl`` module is distributed with the Python standard library, it
+has become the overwhelmingly most-popular method for handling TLS in Python.
+An extraordinary majority of Python libraries, both in the standard library and
+on the Python Package Index, rely on the ``ssl`` module for their TLS
+connectivity.
+
+Unfortunately, the preeminence of the ``ssl`` module has had a number of
+unforseen side-effects that have had the effect of tying the entire Python
+ecosystem tightly to OpenSSL. This has forced Python users to use OpenSSL even
+in situations where it may provide a worse user experience than alternative TLS
+implementations, which imposes a cognitive burden and makes it hard to provide
+"platform-native" experiences.
+
+
+Problems
```

Comment:
I think this section to highlight the fact that the `ssl` module is not an abstract API for implementing TLS (heh), but rather exposes OpenSSL implementation details as it's own.